### PR TITLE
chore: resolved the gap issue when Error shows

### PR DESCRIPTION
**Pull request by pnaik1**
Closes: #993 

## Description
#993 
## How Has This Been Tested?
1.Create a DS Project
2.Create a Model Server
3.Deploy a Model
4.Deploy a 2nd Model with the same name as the first

modal

## Request review criteria:

- [x] The commits have meaningful messages (squashes happen on merge by the bot).
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work
- [ ] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

### CODE REVIEW

**Comment by pnaik1** [comment]
@lucferbux , Thank you , I have made the changes

**Comment by lucferbux** [comment]
Great @pnaik1, one quick nit, dont check `Included tags to the UX team if it was a UI/UX change.` if you are not tagging UX.
That's the checkbox we use in case we need validation from someone from the UX team, if that's needed, you need to tag someone from that team.

**Comment by andrewballantyne** [comment]
In the template it will say `<!--- Closes: #123 -->` -- this is a hint to how you can auto link your PR with the issue you have. Replace `#123` with your issue number and GitHub should auto link your issue to your PR (which will auto-close the issue when the PR merges).

File: frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx
Comment (lucferbux) [inline frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:229]: @DaoDaoNoCode @andrewballantyne I would like to know your opinion on this as you have more patternfly experience here, we try to avoid using custom classes, does that rule apply to patternfly classes too?
If so @pnaik1 we might wanna rework this solution trying to use some [layout](https://www.patternfly.org/v4/layouts/stack) strcuture.

**Review by lucferbux — CHANGES_REQUESTED** [review]

File: frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx
Comment (DaoDaoNoCode) [reply frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:229]: Correct, we should avoid using `className` or custom CSS if possible. Using the Stack layout component here would be a great solution. Another solution is you could wait until the feature freeze this Friday, at that time, the Explanability feature will be merged into main branch, where I created a [DashboardModalFooter](https://github.com/opendatahub-io/odh-dashboard/blob/f/mserving-metrics/frontend/src/concepts/dashboard/DashboardModalFooter.tsx) component, and you can pass it to the `footer` props in your Modal. So you don't need to worry about creating the layout by yourself.

Comment (lucferbux) [reply frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:229]: @DaoDaoNoCode is there any ETA for the [f/mserving-metrics](https://github.com/opendatahub-io/odh-dashboard/tree/f/mserving-metrics) to hit `main`? so we can wait for that or either use any other alternative

Comment (pnaik1) [reply frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:229]: @DaoDaoNoCode the [DashboardModalFooter](https://github.com/opendatahub-io/odh-dashboard/blob/f/mserving-metrics/frontend/src/concepts/dashboard/DashboardModalFooter.tsx) component is not yet merged to the main , so do I need to wait till it gets merged or proceed with the another solution?

Comment (DaoDaoNoCode) [reply frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:229]: Yeah, it was last Friday, but it's a sprint delay, so it would be ~2.5 weeks later until it hits the main branch. You can copy-paste this file into the folder in this PR and update your modal to use it.

Comment (pnaik1) [reply frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:229]: yeah sure @DaoDaoNoCode

Comment (pnaik1) [reply frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx]: I have made the required changes

**Review by lucferbux — APPROVED** [review]
/lgtm

This is going to create a conflict when you guys merge the branch @DaoDaoNoCode @alexcreasy just for you to know, but it's not a big deal, just bear this in mind.

File: frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx
Comment (DaoDaoNoCode) [inline frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:217]: PF offers a prop for Modal named `footer`, can you pass the footer component into it instead of render it in the children? I have an example here https://github.com/opendatahub-io/odh-dashboard/blob/f/mserving-metrics/frontend/src/pages/modelServing/screens/metrics/biasConfigurationModal/ManageBiasConfigurationModal.tsx#L71-L82

Comment (DaoDaoNoCode) [reply frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:217]: And also, in this way, I don't think you need the `Stack` layout in the Modal children anymore to stack the form groups. Just simply put all the sections together, and remove all the `Stack` and `StackItem` components here

Comment (lucferbux) [reply frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:217]: @pnaik1

**Comment by DaoDaoNoCode** [comment]
@pnaik1 Most of the parts look good to me, but I gave 1 bad suggestion, could you add the `Stack` and `StackItem` back? I didn't notice there is a `FormSection` component in the modal, which could create a big gap before it, so the Stack layout could make sure all the sections have proper gaps. Sorry again for the misleading.
Also, looks like your test fails, can you check that too?

**Comment by pnaik1** [comment]
@DaoDaoNoCode @lucferbux I have made the changes , but I wanted to clarify one think , that in [DashboardModalFooter](https://github.com/opendatahub-io/odh-dashboard/blob/f/mserving-metrics/frontend/src/concepts/dashboard/DashboardModalFooter.tsx) Component the Cancel button has variant="link" and in all the other modal the cancel button has variant="secondary' , so may be once the f/mserving-metrics get merged to main everything will be uniform

**Comment by DaoDaoNoCode** [comment]
@pnaik1 Hmm, interesting, let me bring this discussion for the UX to the slack channel.

**Review by DaoDaoNoCode — APPROVED** [review]
lgtm

File: frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx
Comment (andrewballantyne) [inline frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:179]: Leaving the default doesn't cause a reload or anything does it? Seems odd to leave the form wrapper when it's not used. I think it provides styles but we override those styles with `Stack`... perhaps look at removing it?

Comment (DaoDaoNoCode) [reply frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:179]: Hmm, i think if we remove the Form component, will the `FormGroup` and `FormField` under still be the same style? We need the label styles for it

Comment (andrewballantyne) [reply frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:179]: Ah okay, interesting -- I thought the field managed that -- but a form did a gapping -- I could be misremembering.

Comment (DaoDaoNoCode) [reply frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:179]: Double checked, the form group inherits many CSS styles defined in the Form component, so we might still want to keep the Form.

Comment (lucferbux) [reply frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:179]: So we should keep the Form component right?
If so we can integrate it

Comment (lucferbux) [reply frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:179]: @DaoDaoNoCode

Comment (DaoDaoNoCode) [reply frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:179]: I think so.

Comment (lucferbux) [reply frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ManageInferenceServiceModal.tsx:179]: Perfect!

**Review by lucferbux — APPROVED** [review]
/lgtm

**Pull request merged**
